cmd/neofs-node/object.go

```diff
@@ -665,19 +681,73 @@ func (o objectSource) Head(ctx context.Context, addr oid.Address) (*objectSDK.Ob
 }
 
 func (o objectSource) Search(ctx context.Context, cnr cid.ID, filters objectSDK.SearchFilters) ([]oid.ID, error) {
-	var sw searchWriter
+	mProcessedNodes := make(map[string]struct{})
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+
+	result := make([]oid.ID, 0)
+	err := o.containers.ForEachRemoteContainerNode(cnr, func(node netmapsdk.NodeInfo) {
```

Comment:
Not sure why you have to do it this way, it should be a regular SearchV2 call just like it was with the old Search call.